app/ydoc-shared/src/ast/tree.ts

```diff
@@ -770,10 +770,13 @@ export interface MutableStatement extends BaseMutableStatement {
   mutableDocumentationText?: () => Y.Text
 }
 
-abstract class BaseExpression extends Ast {
+/** A base class for all {@link Expression} ASTs */
+export abstract class BaseExpression extends Ast {
```

Comment:
Why export this? Its API is public through the `Expression` interface, which exposes everything except for allowing `instanceof` checks, which shouldn't be used to identify expressions.